src/utils/rdtsc.h

```diff
@@ -117,42 +119,13 @@ static inline tscval_t get_tsc_rate_per_second()
     return tsc_per_second;
 }
 
-/**
- * 'gettimeofday()' based on RDTSC
- * Re-sync with system clock no more then once a second
- */
-inline int gettimefromtsc(struct timespec *ts)
-{
-    static tscval_t tsc_start = TSCVAL_INITIALIZER;
-    static struct timespec ts_start = TIMESPEC_INITIALIZER;
-
-    struct timespec ts_delta = TIMESPEC_INITIALIZER;
-    tscval_t tsc_now, tsc_delta;
-    uint64_t nsec_delta = 0;
-
-    if (!ts_isset(&ts_start)) {
-        clock_gettime(CLOCK_MONOTONIC, &ts_start);
-        gettimeoftsc(&tsc_start);
-    }
-    gettimeoftsc(&tsc_now);
-    tsc_delta = tsc_now - tsc_start;
-    nsec_delta = tsc_delta * NSEC_PER_SEC / get_tsc_rate_per_second();
-
-    ts_delta.tv_sec = nsec_delta / NSEC_PER_SEC;
-    ts_delta.tv_nsec = nsec_delta - ts_delta.tv_sec * NSEC_PER_SEC;
-    ts_add(&ts_start, &ts_delta, ts);
-
-    // Once a second re-sync our start time with real time-of-day
-    if (tsc_delta > get_tsc_rate_per_second()) {
-        ts_clear(&ts_start);
-    }
-
-    return 0;
-}
-
 static inline int gettime(struct timespec *ts)
 {
-    return gettimefromtsc(ts);
+    auto now = steady_clock::now().time_since_epoch();
+    auto now_seconds = duration_cast<seconds>(now);
+    ts->tv_sec = now_seconds.count();
```

Comment:
what happens when tv_sec is only 32 bit field?
happens when "__USE_TIME_BITS64" is not defined

Reply:
In old 32-bit systems where the time_t does not contain at least 35-bits, two things might happen:

- Compiler will warn about the value truncation.
- In runtime, the two instances(start, end) of gettime will still be comparable. Unless a wraparound occurred in runtime(extremely unlikely) there will be no issue.